File: csharp/FizzBuzz/Implementation/Util/Rule.cs
namespace Implementation.Util
{
    public class Rule
    {
        public int Denominator { get; private set; }
        public string Result { get; private set; }

        public Rule(int denominator, string result)
        {
            Denominator = denominator;
            Result = result;
        }
    }
}

File: java/rock-paper-scissors/src/main/java/Rule.java
import java.util.function.BiFunction;

class Rule {
    private final BiFunction<PlayerChoice, PlayerChoice, Boolean> applies;
    private final Result result;

    Rule(BiFunction<PlayerChoice, PlayerChoice, Boolean> applies, Result result) {
        this.applies = applies;
        this.result = result;
    }

    boolean appliesTo(PlayerChoice self, PlayerChoice opponent) {
        return applies.apply(self, opponent);
    }

    Result getResult() {
        return result;
    }
}
